File: erpnext/projects/report/project_update_report/project_update_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
        data    = get_data(filters)
        columns = get_columns()
	return columns, data

def get_data(filters):
        
        data = frappe.db.sql("""
                        select project as activity, posting_date, percent_completed
                        from  `tabAchievement Entry Sheet`  
                        where project    = "{0}"
                        
                        order by posting_date asc 
                        """.format(filters.get("activity")), as_dict=1)

        
        return data
        
def get_columns():
        return [
                {
                        "fieldname": "activity",
                        "label": ("Project"),
                        "fieldtype": "Link",
                        "options": "Project",
                        "width": 400
                },
                {
                        "fieldname": "posting_date",
                        "label": ("Update Date"),
                        "width": 150
                },
                {
                        "fieldname": "percent_completed",
                        "label": ("Percent Updated"),
                        "fieldtype": "float",
                        "width": 100
                },
                # {
                #         "fieldname": "posting_date",
                #         "label": _("Update Date"),
                #         "width": 150
                # },
                # {
                #         "fieldname": "designation",
                #         "label": _("Designation"),
                #         "width": 150
                # },
               
                # {
                #         "fieldname": "branch",
                #         "label": _("Branch"),
                #         "fieldtype": "Link",
                #         "options": "Branch",
                #         "width": 100
                # },
               
        ]

File: erpnext/projects/report/project_update_report/project_update_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Project Update Report"] = {
	"filters": [
		{
			"fieldname": "project",
			"label": ("Gyalsung Academy"),
			"fieldtype": "Link",
			"options": "Project",
			"get_query": function() {
							return {
											'doctype': "Project",
											'filters': [
															['is_group', '=', '1']
											]
							}
			},
	},
	{
			"fieldname": "activity",
			"label": ("Activity"),
			"fieldtype": "Link",
			"options": "Project",
			"get_query": function() {
							var parent_project = frappe.query_report.filters_by_name.project.get_value();
							return { 'doctype': "Project",
									'filters': [
											['is_group', '=', '0'],
											['parent_project', '=', parent_project]
					]
			}
	}
}
               

	]
}

